File: 백준/Silver/2606. 바이러스/바이러스.cc
#include <bits/stdc++.h>
using namespace std;

vector<int> adj[103];
int n, node, u, v, visited[103], num;

int main() {
    cin >> n;
    cin >> node;
    while(node--) {
        cin >> u >> v;
        adj[u].push_back(v);
        adj[v].push_back(u);
    }

    queue<int> q;
    q.push(1);
    visited[1] = 1;
    while(q.size()) {
        int cur = q.front();
        q.pop();
        for(auto nxt : adj[cur]) {
            if(visited[nxt]) continue;
            visited[nxt] = 1;
            num++;
            q.push(nxt);
        }
    }
    cout << num;
}

File: 백준/Silver/2606. 바이러스/바이러스.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {
    static ArrayList<ArrayList<Integer>> graph = new ArrayList<>();
    static boolean[] visited;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int m = Integer.parseInt(br.readLine());

        visited = new boolean[n + 1];

        for (int i = 0; i < n + 1; i++) {
            graph.add(new ArrayList<>());
        }

        for (int i = 0; i < m; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int u = Integer.parseInt(st.nextToken());
            int v = Integer.parseInt(st.nextToken());

            graph.get(u).add(v);
            graph.get(v).add(u);
        }

        Queue<Integer> queue = new LinkedList<>();
        int count = 0;
        queue.add(1);
        visited[1] = true;
        while (!queue.isEmpty()) {
            int cur = queue.poll();
            for (int nxt : graph.get(cur)) {
                if (visited[nxt]) continue;
                queue.add(nxt);
                visited[nxt] = true;
                count++;
            }
        }
        System.out.println(count);

    }
}